pkg/controller/channel/channel_controller.go

```diff
@@ -217,125 +239,119 @@ func (r *ReconcileChannel) Reconcile(request reconcile.Request) (reconcile.Resul
 
 	err = r.validateClusterRBAC(instance)
 	if err != nil {
-		klog.Info("Failed to validate RBAC for clusters for channel ", instance.Name, " with error: ", err)
+		klog.Info("failed to validate RBAC for clusters for channel ", instance.Name, " with error: ", err)
 		return reconcile.Result{}, err
 	}
 
 	// If the channel has relative secret and configMap, annotate the channel info in the secret and configMap
-	if instance.Spec.SecretRef != nil && instance.Spec.SecretRef.Name > "" {
-		secretName := instance.Spec.SecretRef.Name
 
-		secretNamespace := instance.Spec.SecretRef.Namespace
-		if secretNamespace == "" {
-			secretNamespace = instance.Namespace
-		}
-
-		secrectInstance := &corev1.Secret{}
-		secetKey := types.NamespacedName{Name: secretName, Namespace: secretNamespace}
-
-		err = r.Get(context.TODO(), secetKey, secrectInstance)
-		if err == nil {
-			localLabels := secrectInstance.GetLabels()
-			if localLabels == nil {
-				localLabels = make(map[string]string)
-			}
-
-			localLabels[chv1.ServingChannel] = "true"
-			secrectInstance.SetLabels(localLabels)
+	//sync the channel to the serving-channel annotation in all involved secrets.
+	srtRef := instance.Spec.SecretRef
 
-			err = r.Update(context.TODO(), secrectInstance)
-			klog.Infof("Set label serving-channel to secret object: %#v, error: %#v", *secrectInstance, err)
-		}
-		//sync the channel to the serving-channel annotation in all involved secrets.
-		r.syncSecrectAnnotation(instance, request.NamespacedName)
+	if err := r.updatedReferredObjLabel(srtRef, srtGvk); err != nil {
+		klog.Errorf("failed to update referred secret label %v", err)
 	}
 
-	r.updateConfigMap(instance)
+	if err := r.syncReferredObjAnnotation(request, srtRef, srtGvk); err != nil {
+		klog.Errorf("faild to annotation %v", err)
+	}
 
 	//sync the channel to the serving-channel annotation in all involved ConfigMaps.
-	r.syncConfigAnnotation(instance, request.NamespacedName)
+	//r.syncConfigAnnotation(instance, request.NamespacedName)
+	cmRef := instance.Spec.ConfigMapRef
+	if err := r.updatedReferredObjLabel(cmRef, cmGvk); err != nil {
+		klog.Errorf("failed to update referred configMap label %v", err)
+	}
+
+	if err := r.syncReferredObjAnnotation(request, cmRef, cmGvk); err != nil {
+		klog.Errorf("faild to annotation %v", err)
+	}
 
 	return reconcile.Result{}, nil
 }
 
-func (r *ReconcileChannel) updateConfigMap(instance *chv1.Channel) {
-	if instance.Spec.ConfigMapRef != nil && instance.Spec.ConfigMapRef.Name > "" {
-		configName := instance.Spec.ConfigMapRef.Name
-		configNamespace := instance.Spec.ConfigMapRef.Namespace
+func (r *ReconcileChannel) updatedReferredObjLabel(ref *corev1.ObjectReference, objGvk schema.GroupVersionKind) error {
```

Comment:
Just 'updateObjectLabels'?

Reply:
"refer to sth." is not "reference to sth." in English, so ... maybe in the function/var names and comments, we should use "referenced" rather than "referred" to avoid confusion.

Reply:
yup `updatedReferencedObjectLabels` fits better